dev,3.0: Add backup and restore documentation (#1368)
pingcap/docs: dev/how-to/maintain/tidb-in-kubernetes/backup-and-restore.md

    ```

* Run the following command to create an ad-hoc backup job:

    {{< copyable "shell-regular" >}}

    ```shell
    helm install pingcap/tidb-backup --name=<backup-name> --namespace=<namespace> -f values.yaml --version=<tidb-operator-version>
    ```

### View backups

For backups stored in PV, you can view the PVs by using the following command:

{{< copyable "shell-regular" >}}

```shell
kubectl get pvc -n <namespace> -l app.kubernetes.io/component=backup,pingcap.com/backup-cluster-name=<cluster-name>
```

If you store your backup data to [Google Cloud Storage](https://cloud.google.com/storage/), [Ceph Object Storage](https://ceph.com/ceph-storage/object-storage/) or [Amazon S3](https://aws.amazon.com/s3/), you can view the backups by using the GUI or CLI tool of these storage provider.

## Restore

The helm chart `pingcap/tidb-backup` helps restore a TiDB cluster using backup data. Refer to the steps below to perform a restore operation:

* Modify the `values.yaml` file:
    * Set `clusterName` to the target TiDB cluster name.
    * Set `mode` to `restore`.

>     * Set `name` to the backup name you want to restore([view backups](#view-backups) helps you view all the backups available). If the backup is stored in `Google Cloud Storage`, `Ceph Object Storage` or `Amazon S3`, you must configure the corresponding section too (you might continue to use the same configuration you set in the [adhoc full backup](#ad-hoc-full-backup)).

Reviewer: ```suggestion
    * Set `name` to the name of the backup you want to restore (refer to [view backups](#view-backups) to view available backups). If the backup is stored in [Google Cloud Storage](https://cloud.google.com/storage/), [Ceph Object Storage](https://ceph.com/ceph-storage/object-storage/) or [Amazon S3](https://aws.amazon.com/s3/), you must configure the corresponding sections and make sure that the same configurations are applied as you perform the [full backup](#full-backup).
```

> * Create a Kubernetes [Secret](https://kubernetes.io/docs/concepts/configuration/secret/) containing the user and password that has the privilege to restore the database, and set `secretName` in the `values.yaml` file to the name of the created `Secret`(default to `backup-secret`, skip this if you have already created one in the [adhoc full backup](#ad-hoc-full-backup) section):

Reviewer: ```suggestion
* Create a Kubernetes [Secret](https://kubernetes.io/docs/concepts/configuration/secret/) containing the user and password (the user must have the database-related privileges to back up the data). Meanwhile, set `secretName` in the `values.yaml` file to the name of the created `Secret` (default to `backup-secret`; skip this if you have already created one when you perform [full backup](#full-backup)):
```

    {{< copyable "shell-regular" >}}

    ```shell
    kubectl create secret generic backup-secret -n <namespace> --from-literal=user=<user> --from-literal=password=<password>
    ```

* Restore the backup:

    {{< copyable "shell-regular" >}}

    ```shell
    helm install pingcap/tidb-backup --namespace=<namespace> --name=<restore-name> -f values.yaml --version=<tidb-operator-version>
    ```

## Incremental backup

> Incremental backup leverages the [TiDB Binlog](https://www.pingcap.com/docs/dev/reference/tools/tidb-binlog/overview/) tool to collect binlog data from TiDB and provide real-time backup and replication to downstream platforms.

Reviewer: ```suggestion
Incremental backup uses [TiDB Binlog](/dev/reference/tools/tidb-binlog/overview.md) to collect binlog data from TiDB and provide real-time backup and replication to downstream databases.
```

> Incremental backup is disabled in the TiDB cluster by default. To create a TiDB cluster with incremental backup enabled or enable incremental backup in existing TiDB cluster, modify the `values.yaml` file:

Reviewer: ```suggestion
Incremental backup is disabled in TiDB by default. To enable it, modify the `values.yaml` file:
```

* Set `binlog.pump.create` to `true`.
* Set `binlog.drainer.create` to `true`.

> * Set `binlog.pump.storageClassName` and `binlog.drainer.storageClassName` to a proper `storageClass` available in your kubernetes cluster.

Reviewer: ```suggestion
* Set `binlog.pump.storageClassName` and `binlog.drainer.storageClassName` to an available `storageClass` in your Kubernetes cluster.
```

> * Set `binlog.drainer.destDBType` to your desired downstream, explained in detail below.

Reviewer: ```suggestion
* Set `binlog.drainer.destDBType` to your desired downstream storage as needed, which is explained in details below.
```

> Three types of downstream platforms available for incremental backup:

Reviewer: ```suggestion
Incremental backup supports three types of downstream storage:
```

> * PersistenceVolume: default downstream. You can consider configuring a large PV for `drainer` (the `binlog.drainer.storage` variable) in this case.

Reviewer: ```suggestion
* PersistenceVolume: the default downstream storage. You can consider configuring a large PV for `drainer` (by modifying `binlog.drainer.storage`) in this case.
```

> * MySQL compatible database: enabled by setting `binlog.drainer.destDBType` to `mysql`. You must configure the target address and credential in the `binlog.drainer.mysql` section too.

Reviewer: ```suggestion
* MySQL compatible databases: enabled by setting `binlog.drainer.destDBType` to `mysql`. Meanwhile, you must configure the address and credential of the target database in `binlog.drainer.mysql`.
```

> * Apache Kafka: enable by setting `binlog.drainer.destDBType` to `kafka`. You must configure the zookeeper address and Kafka address in the `binlog.drainer.kafka` section too.

Reviewer: ```suggestion
* Apache Kafka: enabled by setting `binlog.drainer.destDBType` to `kafka`. Meanwhile, you must configure the zookeeper address and Kafka address of the target cluster in `binlog.drainer.kafka`.
```
